src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/ASEReadCounter.java

```diff
@@ -0,0 +1,301 @@
+        outputStream.println(header);
+    }
+
+    @Override
+    public void apply(AlignmentContext alignmentContext, ReferenceContext referenceContext, FeatureContext featureContext) {
+        final String contig = alignmentContext.getContig();
+        final long position = alignmentContext.getPosition();
+
+
+        final char refAllele = (char) referenceContext.getBase();
+
+        final List<VariantContext> VCs = featureContext.getValues(variants);
+        if (VCs != null && VCs.size() > 1) {
+            throw new UserException("More then one variant context at position: " + contig + ":" + position);
+        }
+        if (VCs == null || VCs.isEmpty()) {
+            return;
+        }
+
+        final VariantContext vc = VCs.get(0);
+        if (!vc.isBiallelic()) {
+            logger.warn("Ignoring site: cannot run ASE on non-biallelic sites: " + vc.toString());
+            return;
+        }
+
+        if (vc.getHetCount() < 1) {
+            logger.warn("Ignoring site: variant is not het at postion: " + contig + ":" + position);
+            return;
+        }
+
+        if (vc.getNAlleles() == 1 || vc.getAlternateAllele(0).getBases().length == 0) {
+            throw new UserException("The file of variant sites must contain heterozygous sites and cannot be a GVCF file containing <NON_REF> alleles.");
+        }
+
+        final char altAllele = (char) vc.getAlternateAllele(0).getBases()[0];
+
+        final String siteID = vc.getID();
+        final ReadPileup pileup = filterPileup(alignmentContext.getBasePileup(), countType);
+
+        // count up the depths of all and QC+ bases
+        final StringBuilder line = calculateLineForSite(pileup, siteID, refAllele, altAllele);
+        if (line != null) {
+            outputStream.println(line);
+        }
+    }
+
+    @Override
+    public void closeTool() {
+        if (outputStream != null)
+            outputStream.close();
+    }
+
+    private ReadPileup filterPileup(final ReadPileup originalPileup, final CountPileupType countType) {
+
+        final ReadPileup pileupWithDeletions;
+        switch (countType) {
+            case COUNT_FRAGMENTS_REQUIRE_SAME_BASE: pileupWithDeletions = originalPileup.getOverlappingFragmentFilteredPileup(true, true);
+                break;
+            case COUNT_READS: pileupWithDeletions = originalPileup;
+                break;
+            case COUNT_FRAGMENTS: pileupWithDeletions = originalPileup.getOverlappingFragmentFilteredPileup(false, true);
+                break;
+            default: throw new UserException("Must use valid CountPileupType");
+        }
+
+        return pileupWithDeletions.makeFilteredPileup(p -> !p.isDeletion());
+
+    }
+
+    private StringBuilder calculateLineForSite(final ReadPileup pileup, final String siteID, final char refAllele, final char altAllele) {
```

Comment:
I would return the String that gets build and not the builder itself.  Returning a `StringBuilder` sort of implies that the builder is intended to be used in some way, i.e. to continue building a line.  

In the no output case, you could return an empty string, and then print that instead of println if the new line is the issue.  Or just check the null case and not do anything like you're doing now.  

In general, my feeling is that void methods are the devil and should be as absolutely clear in the function as possible.